src/main/java/lt/vu/web/api/v1/factory/AddressFactory.java

```diff
@@ -10,10 +10,19 @@
 
     public Address create(PostAddressDTO addressDTO) {
         Address address = new Address();
+        validate(addressDTO);
 
         address.setCity(addressDTO.getCity());
         address.setStreet(addressDTO.getStreet());
 
         return address;
     }
+
+    private void validate(PostAddressDTO addressDTO){
```

Comment:
I don't think validation logic belongs inside a factory. Maybe a validator service used in controller would be better?